src/main/java/com/spotify/docker/client/DefaultDockerClient.java

```diff
@@ -2647,16 +2648,40 @@ public RegistryAuth registryAuth() {
      *
      * @param registryAuth RegistryAuth object
      * @return Builder
+     *
+     * @deprecated in favor of {@link #registryAuthSupplier(RegistryAuthSupplier)}
      */
+    @Deprecated
     public Builder registryAuth(final RegistryAuth registryAuth) {
+      if (this.registryAuthSupplier != null) {
+        throw new IllegalStateException(ERROR_MESSAGE);
+      }
       this.registryAuth = registryAuth;
+
+      // stuff the static RegistryAuth into a RegistryConfigs instance to maintain what
+      // DefaultDockerClient used to do with the RegistryAuth before we introduced the
+      // RegistryAuthSupplier
+      final RegistryConfigs configs = RegistryConfigs.create(singletonMap(
+          MoreObjects.firstNonNull(registryAuth.serverAddress(), ""),
+          registryAuth
+      ));
+
+      this.registryAuthSupplier = new NoOpRegistryAuthSupplier(registryAuth, configs);
+      return this;
+    }
+
+    public Builder registryAuthSupplier(final RegistryAuthSupplier registryAuthSupplier) {
+      if (this.registryAuthSupplier != null) {
```

Comment:
Shouldn't this check `if(this.registryAuth != null)`? Right now this would only get triggered if the user called `builder().registryAuthSupplier(registryAuthSupplier).registryAuthSupplier(registryAuthSupplier)`.